core/include/kernel/tcg.h

```diff
@@ -0,0 +1,163 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ *
+ * This file refers the following TCG specification.
+ * TCG PC Client Platform Firmware Profile Specification
+ */
+
+#ifndef __KERNEL_TCG_H__
+#define __KERNEL_TCG_H__
+
+#include <tee_api_types.h>
+#include <tpm2.h>
+
+#define TPM2_EVENT_LOG_SIZE		4096
+
+/*
+ * SHA1 Event Log Entry Format
+ *
+ * @pcr_index:  PCRIndex event extended to
+ * @event_type: Type of event (see EFI specs)
+ * @digest:     Value extended into PCR index
+ * @event_size: Size of event
+ * @event:      Event data
+ */
+struct tcg_pcr_event {
+	uint32_t pcr_index;
+	uint32_t event_type;
+	uint8_t digest[TPM2_SHA1_DIGEST_SIZE];
+	uint32_t event_size;
+	uint8_t event[];
+} __packed;
```

Comment:
The `__packed` attributes make access to the struct inefficient since it also means that the compiler should generate code for unaligned access. In some cases it may be a reasonable compromise,  but is it really needed here?

Reply:
Since digest arrays size is fixed to 20 bytes, not having this __packed won;t have any impact. I will remove it